benchmark/simulate_hgts.py

```diff
@@ -0,0 +1,627 @@
+    log_f.write("#type\tdonor\tstart\tend\trecipient\tstart\t"
+                "end\tstrand\n")
+    # begin simulation
+    for x in range(0, num_hgts):
+        # select random donor gene (for HGT)
+        gene_donor_label = gene_donor_labels[x]
+        idx_recip = gene_positions_s[idx[x]][1] + 1
+        # beginning from valid position for inserting new gene, check whether
+        # the length can fit without overlapping with existing gene, otherwise
+        # search for next valid position
+        for y in range(idx[x], len(gene_positions_s)-1):
+            if idx_recip + len(genes_donor[gene_donor_label][0])*3 <\
+                    gene_positions_s[y+1][0]:
+                idx_end = idx_recip + len(genes_donor[gene_donor_label][0])*3
+                # insert gene (protein)
+                hgt_gene = "%s_hgt_n" % gene_donor_label
+                genes_recip[hgt_gene] =\
+                    [genes_donor[gene_donor_label][0], idx_recip, idx_end,
+                     genes_donor[gene_donor_label][3]]
+                # insert gene (nucleotide)
+                seq_recip = Sequence(
+                    str(seq_recip[:idx_recip]) +
+                    str(seq_donor[genes_donor[gene_donor_label][1]:
+                        genes_donor[gene_donor_label][2]]) +
+                    str(seq_recip[idx_recip:]))
+                # write HGTs to log file
+                log_f.write(
+                    "n\t%s\t%s\t%s\t%s\t%s\t%s\t%s\n" %
+                    (gene_donor_label,
+                     genes_donor[gene_donor_label][1],
+                     genes_donor[gene_donor_label][2],
+                     hgt_gene,
+                     idx_recip,
+                     idx_end,
+                     genes_donor[gene_donor_label][3]))
+                break
+            # try next open region
+            idx_recip = gene_positions_s[y+1][1] + 1
+
+    return seq_recip
+
+
+def write_results(genes_donor,
+                  donor_genome_fp,
+                  genes_recip,
+                  recip_genome_fp,
+                  seq_donor,
+                  seq_recip,
+                  output_dir):
+    """Write donor and recipient genomes to FASTA files (nucl and protein).
+
+    Parameters
+    ----------
+    genes_donor: dictionary
+    seq_donor: skbio.sequence.Sequence
+        Sequence object for donor genome
+    genes_recip: dictionary
+    seq_recip: skbio.sequence.Sequence
+        Sequence object for recipient genome
+    """
+    # output dir for simulated results
+    simulated_dir = join(output_dir, "simulated")
+    if not exists(simulated_dir):
+        makedirs(simulated_dir)
+    donor_genome_aa_fp = join(
+        simulated_dir, "%s.faa" % basename(splitext(donor_genome_fp)[0]))
+    recip_genome_aa_fp = join(
+        simulated_dir, "%s.faa" % basename(splitext(recip_genome_fp)[0]))
+    with open(donor_genome_aa_fp, 'w') as donor_genome_aa_f:
+        for gene in genes_donor:
```

Comment:
why are these manually formatted whereas the nucleotide data is using the skbio FASTA formatter?